app/models/events/review.rb

```diff
@@ -35,5 +37,29 @@ class Review < ApplicationRecord
 
     validates :status, inclusion: { in: statuses.keys }
     validates :github_id, presence: true
+
+    ##
+    # Returns the occurrence time of the 'review' event.
+    ##
+    def occurence_time(payload:)
+      event_action = payload.fetch('action')
+      review_payload = payload.fetch('review')
+      case event_action
+      when 'submitted'
+        parse_time(review_payload.fetch('submitted_at'))
+      when 'edited'
+        parse_time(review_payload.fetch('edited_at'))
+      when 'dismissed'
+        parse_time(review_payload.fetch('dismissed_at'))
+      else
+        raise "Uknown action '#{event_action}'"
```

Comment:
We should not raise errors. As Santi told, this is a job, whenever it raises an error it will retry, which will come to an infinity loop.

Reply:
I think we should configure the maximum number of retries to avoid infinite loops.
I don't think it is possible to rely on a Job not raising errors during its execution. Runtime errors can be raised for reasons not in control of the Job (find! or create! failures, lost connections to databases or queues, unexpected or inconsistent data in a records).